src/test/unit_tests/test_nc_online_upgrade_s3_integrations.js

```diff
@@ -0,0 +1,190 @@
+        } catch (err) {
+            console.log('test_nc_online_upgrade_s3_integrations: failed deleting default bucket', err);
+        }
+    });
+
+    mocha.it('create bucket - host is blocked for config dir updates - should fail', async function() {
+        await update_system_json(NC_CORETEST_CONFIG_FS, mock_old_config_dir_version);
+        await assert.rejects(async () => s3_client.createBucket({ Bucket: bucket_defaults.name }), err => {
+            assert.strictEqual(err.Code, 'InternalError');
+            return true;
+        });
+    }, TEST_TIMEOUT);
+
+    mocha.it('create bucket - host is not blocked for config dir updates', async function() {
+        await update_system_json(NC_CORETEST_CONFIG_FS);
+        await s3_client.createBucket({ Bucket: bucket_defaults.name });
+    }, TEST_TIMEOUT);
+
+    mocha.it('put bucket policy - host is blocked for config dir updates - should fail', async function() {
+        await create_default_bucket(s3_client, bucket_defaults.name);
+        await update_system_json(NC_CORETEST_CONFIG_FS, mock_old_config_dir_version);
+        await assert.rejects(async () => s3_client.putBucketPolicy({ Bucket: bucket_defaults.name, Policy: JSON.stringify(bucket_policy) }),
+            err => {
+                assert.strictEqual(err.Code, 'InternalError');
+                return true;
+            }
+        );
+    }, TEST_TIMEOUT);
+
+    mocha.it('put bucket policy - host is not blocked for config dir updates', async function() {
+        await create_default_bucket(s3_client, bucket_defaults.name);
+        await update_system_json(NC_CORETEST_CONFIG_FS);
+        await s3_client.putBucketPolicy({ Bucket: bucket_defaults.name, Policy: JSON.stringify(bucket_policy) });
+    }, TEST_TIMEOUT);
+
+    mocha.it('delete bucket - host is blocked for config dir updates - should fail', async function() {
+        await create_default_bucket(s3_client, bucket_defaults.name);
+        await update_system_json(NC_CORETEST_CONFIG_FS, mock_old_config_dir_version);
+        await assert.rejects(async () => s3_client.deleteBucket({ Bucket: bucket_defaults.name }), err => {
+            assert.strictEqual(err.Code, 'InternalError');
+            return true;
+        });
+    }, TEST_TIMEOUT);
+
+    mocha.it('delete bucket - host is not blocked for config dir updates', async function() {
+        await create_default_bucket(s3_client, bucket_defaults.name);
+        await update_system_json(NC_CORETEST_CONFIG_FS);
+        await s3_client.deleteBucket({ Bucket: bucket_defaults.name });
+    }, TEST_TIMEOUT);
+
+    mocha.it('head bucket', async function() {
+        await create_default_bucket(s3_client, bucket_defaults.name);
+        await update_system_json(NC_CORETEST_CONFIG_FS, mock_old_config_dir_version);
+        await s3_client.headBucket({ Bucket: bucket_defaults.name });
+    }, TEST_TIMEOUT);
+
+    mocha.it('PUT object - host is blocked for config dir updates', async function() {
```

Comment:
only bucket specific actions will get restricted while upgrading, not the object operations, Is that right?

Reply:
create/update/delete bucket ops are restricted, 
list/get/head bucket is allowed
object operations are allowed